thepracticaldev/dev.to
# Fetch article by slug

## What type of PR is this? (check all applicable)


## Description
According to this issue #8313 the user suggested that articles should be searchable by slug and not only by id. My implement included converting the param input to an integer, so if a string that is not a valid number is passed it returns 0. So if the value is not zero then I find with the id else I find by the slug,  in both cases, the error is well handled by rails, this is why I used `find_by!` instead of `find_by` because the former return an ` ActiveRecord::ActiveRecordError` when nothing is found

## QA Instructions, Screenshots, Recordings
This is an example of a working slug
![workingslug](https://user-images.githubusercontent.com/37092867/85867764-40165d00-b7c1-11ea-9bd6-9d9ebea37c66.png)

Example of a wrong slug that is not found 
![wrongslug](https://user-images.githubusercontent.com/37092867/85867802-51f80000-b7c1-11ea-824e-1b0ce6242eb7.png)

Also, the id works fine
![id](https://user-images.githubusercontent.com/37092867/85868232-ebbfad00-b7c1-11ea-8fd6-43beb0e04796.png)

I tested everything, all possible endpoints and they work fine. 

I am opening this as a draft PR to check if this is a good approach, if this is okay I can update the documentation.

_Please replace this line with instructions on how to test your changes, as well
as any relevant images for UI changes._

## Added tests?


## Added to documentation?
